packages/atomic/src/components/commerce/atomic-commerce-search-box-query-suggestions/molecule-2.spec.ts

```diff
@@ -0,0 +1,69 @@
+import {buildFakeLoadQuerySuggestActions} from '@/vitest-utils/testing-helpers/fixtures/headless/commerce/query-suggest-actions';
+import {loadQuerySuggestActions} from '@coveo/headless/commerce';
+import {describe, it, vi, beforeEach} from 'vitest';
+import './atomic-commerce-search-box-query-suggestions';
+
+vi.mock('@coveo/headless/commerce', {spy: true});
+
+/**
+ * This way adds does not add describe blocks for each lifecycles of the component since these are
+ * theoretically private and not called directly by us or the user.
+ *
+ * It adds a describe block for the initialize function since it is a public function that is called
+ * by the atomic-commerce-search-box.
+ *
+ * This way, we don't mention private terms like 'connected' or 'updated' or 'initialized'.
+ * In the general describe we test everything from the lifecycle to the rendering. We make more
+ * describes for public functions that are called by the user or us.
+ */
+
+describe('AtomicCommerceSearchBoxQuerySuggestions', () => {
+  beforeEach(() => {
+    vi.mocked(loadQuerySuggestActions).mockReturnValue(
+      buildFakeLoadQuerySuggestActions()
+    );
+  });
+
+  describe('when outside of a search box', () => {
+    it('should log an error in the console', async () => {});
+
+    it('should display an error component', async () => {});
+  });
+
+  describe('when inside a search box', () => {
+    it('should dispatch the search box suggestions event', async () => {});
+
+    it('should assign the bindings from the event', async () => {});
+
+    it('should call initialize after assigning bindings', async () => {});
```

Comment:
Here, the `initialize` method could be spied on & asserted on because it would be tested in `#initialize`